metricbeat/module/prometheus/query/config.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package query
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// Config for "query" metricset
+type Config struct {
```

Comment:
nit: "What's a `Config` for a query in prometheus Beats code?" If it is a "bunch of Path, QueryParams and QueryNames" it doesn't sound like a config of kind "Raise the size of a shard, set this parameter to `true` or set the heap to blah". Maybe the struct must be called `Queries` and the field inside the struct `UserSpecifiedQueries` (I don't know if that `[]QueryConfig` represents user specified queries, in fact).

BTW, I'm very extreme in this and if I find that I cannot find a good name for a struct I think that the entire idea is wrong and I refactor code until I find good naming for evertything... but that's my particular obsessive compulsive disorder :smile: